### ISSUE TITLE
Unable to build using opam 2.1.3

### ISSUE THREAD

**Issue by quentinschibler** [ott-lang/ott#94]
Trying to build ott 0.32 using opam result in the following error :

```
#=== ERROR while compiling ott.0.32 ===========================================#
# context     2.1.3 | linux/x86_64 | ocaml.4.14.0 | https://opam.ocaml.org#797a89bb
# path        ~/.opam/default/.opam-switch/build/ott.0.32
# command     /usr/bin/make world
# exit-code   2
# env-file    ~/.opam/log/ott-8726-fc33d7.env
# output-file ~/.opam/log/ott-8726-fc33d7.out
### output ###
# +  let s = Buffer.create n in
# [...]
# -    if unsafe_get v i then s.[i] <- '1'
# +    Buffer.add_char s (if unsafe_get v i then '1' else '0')
#    done;
# -  s
# +  Buffer.contents s
#  
#  let print fmt v = Format.pp_print_string fmt (to_string v)
#  
```
I don't really know anything about ocaml, but if you need any other information, feel free to ask

**Comment by bacam** [comment]
I just tried this with the same versions of ocaml and opam without any problems.  Could you send the `output-file` that the error mentions?  The snippet that opam included at the bottom is too short to show the actual error message.

**Comment by quentinschibler** [comment]
I must have been tired not to see that there was actually some logs... It is actually an issue with a patch. 

```
cd ../ocamlgraph-1.7 && patch -p1 < ../ocamlgraph-safe-string.patch
patching file lib/bitv.ml
Hunk 1 FAILED 449/449.
 
 let to_string v =
   let n = v.length in
-  let s = String.make n '0' in
+  let s = Buffer.create n in
   for i = 0 to n - 1 do
-    if unsafe_get v i then s.[i] <- '1'
+    Buffer.add_char s (if unsafe_get v i then '1' else '0')
   done;
-  s
+  Buffer.contents s
 
 let print fmt v = Format.pp_print_string fmt (to_string v)
```

I haven't looked at the file yet, but I am using busybox patch (as opposed to gnu patch), I suspect this is the source of the error

**Comment by bacam** [comment]
Yes, I've made a pull request updating the patch so that it doesn't need gnu patch's fuzzing matching.  You should be able to try it out with
```
opam pin add ott https://github.com/bacam/ott.git#busybox-patch-fix
```

**Comment by quentinschibler** [comment]
Worked like a charm, thank you !

**Comment by bacam** [comment]
Fixed properly by #96, which uses the packaged version of ocamlgraph.
